src/FINDOLOGIC/Api/RequestBuilders/Json/SuggestionRequestBuilder.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace FINDOLOGIC\Api\RequestBuilders\Json;
+
+use FINDOLOGIC\Api\Config;
+use FINDOLOGIC\Api\Definitions\Endpoint;
+use FINDOLOGIC\Api\Definitions\QueryParameter;
+use FINDOLOGIC\Api\Exceptions\InvalidParamException;
+use FINDOLOGIC\Api\RequestBuilders\RequestBuilder;
+use FINDOLOGIC\Api\ResponseObjects\Json\JsonResponse;
+use FINDOLOGIC\Api\Validators\ParameterValidator;
+
+class SuggestionRequestBuilder extends RequestBuilder
+{
+    protected $endpoint = Endpoint::SUGGESTION;
+
+    public function __construct(Config $config)
+    {
+        parent::__construct($config);
+        $this->addRequiredParam(QueryParameter::QUERY);
+    }
+
+    /**
+     * @inheritdoc
+     * @return JsonResponse
+     */
+    public function sendRequest()
+    {
+        $this->checkRequiredParamsAreSet();
+
+        $responseContent = $this->client->request($this->buildRequestUrl());
+        return new JsonResponse($responseContent, $this->client->getResponseTime());
+    }
+
+    /**
+     * Sets the query param. It is used as the search query. When doing a suggestion request the min length for a
+     * query is one character.
+     *
+     * @param $value string
+     * @see https://docs.findologic.com/doku.php?id=smart_suggest_new#request
+     * @return $this
+     */
+    public function setQuery($value)
```

Comment:
All Smart Suggest parameter types are included in `QueryParameter`, but not all of them seem to have a setter. This is inconsistent. What's the reason behind it?

Reply:
The reason for that is that some parameters are ignored/irrelevant for doing suggest requests. It does not make sense to set for example the `attrib` param, because it is completely irrelevant for suggest requests.